### PR TITLE
Enhancement: Easy way to create validation rules with a custom FieldType MetaDataAttribute

### PR DESCRIPTION
**Pull request by clangelov**
GraphQL.Net allow us to plug in the execution process and to add our custom ValidationRules

Inside them we have access to the FieldType object, and the easiest way to interact it is by checking its metadata.
Unfortunately currently we don’t add any Metadata to this particular type.

 This can be changed by adding an abstract Attribute that would provide  such in information an this can be FieldTypeMetaDataAttribute

And then in the process of building the FieldType object to check for ancestors of the FieldTypeMetaDataAttribute and fill the information in the Metadata (see GraphTypeAdapter.cs)

For a simple example check the tests

### CODE REVIEW

File: test/GraphQL.Conventions.Tests/Attributes/MetaData/FieldTypeMetaDataAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        private async Task<ExecutionResult> Resolve_Query(string selectedFileds = "testMethod noAttribute name", params string[] accessPermissions)
+        {
+            var engine = GraphQLEngine
+                .New<CustomAttribute_Query>();
+
+            var user = new TestUserContext(accessPermissions);
+
+            var result = await engine
+                .NewExecutor()
+                .WithQueryString("query { node { " + selectedFileds + " } }")
```
Comment (tlil) [inline test/GraphQL.Conventions.Tests/Attributes/MetaData/FieldTypeMetaDataAttributeTests.cs]:
```suggestion
                .WithQueryString("query { node { " + selectedFields + " } }")
```

**Review by tlil — APPROVED** [review]
Thanks for this, @clangelov . Just a nitpick spelling mistake of `fields` :) Otherwise, I'm fine with the change.

**Comment by clangelov** [comment]
> Thanks for this, @clangelov . Just a nitpick spelling mistake of `fields` :) Otherwise, I'm fine with the change.

@tlil  The typo is fixed so the PR is ready to be merged 😃 🎉

**Comment by tlil** [comment]
Thanks, @clangelov - apologies for the delay. :-)

**Pull request merged**